### PR TITLE
Get/api/users

### PR DESCRIPTION
**Pull request by goldfishdolphin**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -33,4 +33,67 @@ describe('api/topics', () => {
...
+                        expect(user).toHaveProperty('username');
+                        expect(user).toHaveProperty('name');
+                        expect(user).toHaveProperty('avatar_url');
+                    });
+                });
+        });
+        test('404: responds with an error message when passes a route that does not exist ', () => {
+            return request(app)
+                .get('/api/NotARoute')
+                .expect(404)
```
Comment (Rayhaan2k9) [inline __tests__/app.test.js:94]: This test should be in its own block as it applies to any incorrect path :)

File: controllers/get-articleById-controller.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+const { selectArticleById } = require("../models/get-articleById.model");
+exports.getArticleById = (req, res, next) => {
+    const { article_id } = req.params;
+    selectArticleById(article_id).then((article) => {
+        res.status(200).send({ article });
```
Comment (Rayhaan2k9) [inline controllers/get-articleById-controller.js]: It's ok to separate controllers into their own files but you don't have to separate every single one. You could have a controllers file for articles where any articles controllers live and then one for users etc :)

**Review by Rayhaan2k9 — CHANGES_REQUESTED** [review]
Just a couple of little things to fix on this one Naureen. I've left comments for you. It's looking great though! 😄

**Review by Augs0 — APPROVED** [review]
Looking good :D i would just suggest another name change for the controllers / models. We will be handling other requests than get requests, so calling these things like get-users is not going to be very clear later. A more general name like users-controller would be better. This is something you can add in your next PR :) don't forget to get all your PRs into main and to pull down before continuing

**Pull request merged**